indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -636,17 +644,31 @@ def _process(self):
         return jsonify_data(**self.list_generator.render_list())
 
 
-class RHRegistrationsModifyStatus(RHRegistrationsActionBase):
-    """Accept/Reject selected registrations."""
+class RHRegistrationsApprove(RHRegistrationsActionBase):
+    """Accept selected registrations from registration list."""
 
     def _process(self):
-        approve = request.form['flag'] == '1'
         for registration in self.registrations:
-            _modify_registration_status(registration, approve)
-        flash(_("The status of the selected registrations was updated successfully."), 'success')
+            _modify_registration_status(registration, approve=True)
+        flash(_("The selected registrations was successfully '{}'.").format('approved'), 'success')
         return jsonify_data(**self.list_generator.render_list())
 
 
+class RHRegistrationsReject(RHRegistrationsActionBase):
+    """Reject selected registrations from registration list."""
+
+    def _process(self):
+        form = RejectRegistrantsForm(obj=FormDefaults(registration_id=request.args.getlist('registration_id')))
```

Comment:
I think this works as well:

```suggestion
        form = RejectRegistrantsForm(registration_id=request.args.getlist('registration_id'))
```

At least that's what we do e.g. in `BulkAbstractJudgmentForm` which works very similarly (form opened with a list of selected items).

Reply:
Also, this should be `request.form`, not `request.args`. Use `data-method="POST"` in the HTML to load the form with POST as well.

To know whether the form was actually submitted or not, add this code:

```python
    submitted = HiddenField()

    ...

    def is_submitted(self):
        return super().is_submitted() and 'submitted' in request.form
```

You can have a look at `BulkAbstractJudgmentForm` to see a place where this is already done as well :)